### PR TITLE
soundmodem: init at 0.20

### PR DESCRIPTION
**Pull request by ymarkus**
###### Motivation for this change
#108545 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/soundmodem/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+stdenv.mkDerivation rec {
+  name = "soundmodem";
+  version = "0.20";
+
+  src = fetchurl {
+    url = "https://archive.org/download/${name}-${version}/${name}-${version}.tar.gz";
+    sha256 = "156l3wjnh5rcisxb42kcmlf74swf679v4xnj09zy5j74rd4h721z";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/soundmodem/default.nix:12]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/applications/radio/soundmodem/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, pkgconfig, alsaLib, audiofile, gtk2, libxml2 }:
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/soundmodem/default.nix:1]:
```suggestion
{ stdenv, fetchurl, pkg-config, alsaLib, audiofile, gtk2, libxml2 }:
```

File: pkgs/applications/radio/soundmodem/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        - No fixed relationship between bitrate, sampling rate, and modem parameters. Modems may be parametrized, and multiple modems may even run on the same audio channel!
+        - Usermode solution allows the use of MMX, VIS, Floating point and other media instruction sets to speed up computation.
+        - Cross platform builds from a single source code provides ubiquitous availability.
+    '';
+    #homepage = "http://gna.org/projects/soundmodem"; # official, but "Connection refused"
+    homepage = "http://soundmodem.vk4msl.id.au/";
+    downloadPage = "https://archive.org/download/${name}-${version}/${name}-${version}.tar.gz";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ ymarkus ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/soundmodem/default.nix:34]:
```suggestion
```

**Comment by ymarkus** [comment]
Thanks for the input. I've changed the two occurrences. 
Do you mind explaining what the difference is, please? It builds with both so they must be basically the same, no?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108570` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>soundmodem</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
> Thanks for the input. I've changed the two occurrences.
> Do you mind explaining what the difference is, please? It builds with both so they must be basically the same, no?

pkgconfig is an alias for pkg-config and shadowed in buildPythonPackage. We try to get rid of it slowly that we can remove the alias in the future.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108570` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>soundmodem</li>
  </ul>
</details>

**Pull request merged**